online-workflow/src/main/java/eu/itesla_project/online/db/OnlineDbMVStore.java

```diff
@@ -289,35 +289,31 @@ public String getCsvMetrics(String workflowId, OnlineStep step) {
                         paramsIndexes.put(parameter, i);
                         i++;
                     }
-                    cvsWriter.writeRecord(headers);
-                    // write step general metrics, if stored
+                    retTable.add(headers);
+                    // gets step general metrics, if stored
                     if (stepStatesMap.containsKey("_")) {
-                        //LOGGER.debug("Preparing CSV data for wf {} and step {} - general step metrics", workflowId, step.name());
-                        String[] values = getStoredMapValues(wfMVStore, "_", step, stepParamsMap.keySet().size(), paramsIndexes);
-                        cvsWriter.writeRecord(values);
-                    }
-                    // write step metrics for each state, if stored
-                    for (String stateId : stepStatesMap.keySet()) {
-                        if (!"_".equals(stateId)) {
-                            //LOGGER.debug("Preparing CSV data for wf {} and step {} - state {} metrics", workflowId, step.name(), stateId);
-                            String[] values = getStoredMapValues(wfMVStore, stateId, step, stepParamsMap.keySet().size(), paramsIndexes);
-                            cvsWriter.writeRecord(values);
-                        }
+                        retTable.add(getStoredMapValues(wfMVStore, "_", step, stepParamsMap.keySet().size(), paramsIndexes));
                     }
+                    // gets step metrics for each state, if stored
+                    stepStatesMap.keySet().stream()
+                            .filter(x -> (!"_".equals(x)))
+                            .sorted(Comparator.comparing(Integer::valueOf))
+                            .forEach(stateId -> {
+                                retTable.add(getStoredMapValues(wfMVStore, stateId, step, stepParamsMap.keySet().size(), paramsIndexes));
+                            });
                 }
-            } catch (IOException e) {
+            } catch (Exception e) {
                 String errorMessage = "error getting cvs data for step " + step.name() + " and wf id " + workflowId;
                 LOGGER.error(errorMessage);
                 throw new RuntimeException(errorMessage);
```

Comment:
I think it's better to keep the cause of the exception: maybe you should use "throw new RuntimeException(errorMessage, e);"